packages/eip-1193-provider/src/errors.ts

```diff
@@ -0,0 +1,114 @@
+import { ProviderRpcError } from 'viem';
+import { PROVIDER_ERROR_CODE, PROVIDER_ERROR_MESSAGE } from './constants';
+
+class UnrecognizedChainError extends ProviderRpcError {
+  constructor(chainId: string) {
+    const errorMessage = `${PROVIDER_ERROR_MESSAGE.UNRECOGNIZED_CHAIN_ID} Received chainId: ${chainId}`;
+    super(new Error(errorMessage), {
+      code: 4902,
```

Comment:
curious about this code (and the other ones below). Are they defined in the standard?

Reply:
Uhm yes and no lol. I followed [Metamask's JSON-RPC](https://docs.metamask.io/wallet/reference/wallet_switchethereumchain/) spec since they are usually the ones writing and pushing all of the eips.
This particular error is for [EIP-3326](https://eips.ethereum.org/EIPS/eip-3326) but they don't document the errors there but rather in [Metamask's JSON-RPC documentation](https://docs.metamask.io/wallet/reference/wallet_switchethereumchain/).

The whole provider adhere's closely to EIP-1193 but there are these other EIPs that you have to chase down to get the full spec:

https://eips.ethereum.org/EIPS/eip-2255
https://eips.ethereum.org/EIPS/eip-3085
https://eips.ethereum.org/EIPS/eip-3326
https://eips.ethereum.org/EIPS/eip-1474
https://eips.ethereum.org/EIPS/eip-695
https://eips.ethereum.org/EIPS/eip-1102